aws/resource_aws_glue_trigger.go

```diff
@@ -0,0 +1,457 @@
+
+	log.Printf("[DEBUG] Deleting Glue Trigger: %s", d.Id())
+	err := deleteGlueTrigger(conn, d.Id())
+	if err != nil {
+		return fmt.Errorf("error deleting Glue Trigger (%s): %s", d.Id(), err)
+	}
+
+	log.Printf("[DEBUG] Waiting for Glue Trigger (%s) to delete", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{glue.TriggerStateDeleting},
+		Target:  []string{""},
+		Refresh: resourceAwsGlueTriggerRefreshFunc(conn, d.Id()),
+		Timeout: d.Timeout(schema.TimeoutDelete),
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+			return nil
+		}
+		return fmt.Errorf("error waiting for Glue Trigger (%s) to delete", d.Id())
+	}
+
+	return nil
+}
+
+func resourceAwsGlueTriggerRefreshFunc(conn *glue.Glue, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		output, err := conn.GetTrigger(&glue.GetTriggerInput{
+			Name: aws.String(name),
+		})
+		if err != nil {
+			return output, "", err
+		}
+
+		if output.Trigger == nil {
+			return output, "", nil
+		}
+
+		return output, aws.StringValue(output.Trigger.State), nil
+	}
+}
+
+func deleteGlueTrigger(conn *glue.Glue, Name string) error {
+	input := &glue.DeleteTriggerInput{
+		Name: aws.String(Name),
+	}
+
+	_, err := conn.DeleteTrigger(input)
+	if err != nil {
+		if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+			return nil
+		}
+		return err
+	}
+
+	return nil
+}
+
+func expandGlueActions(l []interface{}) []*glue.Action {
+	actions := []*glue.Action{}
+
+	for _, mRaw := range l {
+		m := mRaw.(map[string]interface{})
+
+		action := &glue.Action{
+			JobName: aws.String(m["job_name"].(string)),
+		}
+
+		argumentsMap := make(map[string]string)
+		for k, v := range m["arguments"].(map[string]interface{}) {
+			argumentsMap[k] = v.(string)
+		}
+		action.Arguments = aws.StringMap(argumentsMap)
+
+		if v, ok := m["timeout"]; ok && v.(int) > 0 {
+			action.Timeout = aws.Int64(int64(v.(int)))
+		}
+
+		actions = append(actions, action)
+	}
+
+	return actions
+}
+
+func expandGlueConditions(l []interface{}) []*glue.Condition {
+	conditions := []*glue.Condition{}
+
+	for _, mRaw := range l {
+		m := mRaw.(map[string]interface{})
+
+		condition := &glue.Condition{
+			JobName:         aws.String(m["job_name"].(string)),
+			LogicalOperator: aws.String(m["logical_operator"].(string)),
+			State:           aws.String(m["state"].(string)),
+		}
+
+		conditions = append(conditions, condition)
+	}
+
+	return conditions
+}
+
+func expandGluePredicate(l []interface{}) *glue.Predicate {
+	m := l[0].(map[string]interface{})
+
+	predicate := &glue.Predicate{
+		Conditions: expandGlueConditions(m["conditions"].([]interface{})),
+	}
+
+	if v, ok := m["logical"]; ok && v.(string) != "" {
+		predicate.Logical = aws.String(v.(string))
+	}
+
+	return predicate
+}
+
+func flattenGlueActions(actions []*glue.Action) []interface{} {
+	l := []interface{}{}
+
+	for _, action := range actions {
```

Comment:
can we add some crash protection here incase actions is nil?

Reply:
I do not believe anything is required here: https://play.golang.org/p/-ShUuLbE9cc